### PR TITLE
"byte" formatter

### PR DESCRIPTION
**Pull request by hrbrmstr**
Five new functions, 
- `byte_format` (convert numeric vector to byte measurement labels such as 1024 => "1 Kb")
- `bytes` (the one that does all the work)
- 3 "shorthand" functions: `Kb`, `Mb` & `Gb`

These support binary (1024-base) and SI (1000-base) conversions.

This will enable users to avoid rolling their own or adding `gdata` as a dependency and make "cyber" a first-class citizen since pretty much everyone uses `scales` :-) 

If it'd be better without the three shorthand functions I can remove them and do a new PR.

### CODE REVIEW

File: R/formatter.r [hunk 1]
```diff
@@ -121,6 +121,89 @@ percent_format <- function() {
...
+#' @export
+#' @rdname byte_format
+Gb <- byte_format("Gb", "binary")
+
+#' @export
+#' @rdname byte_format
+bytes <- function(x, symbol="auto", units=c("binary", "si")) {
+
+  symbol <- match.arg(symbol, c("auto",
+        "b",  "Kb",  "Mb",  "Gb",  "Tb",  "Pb",  "Eb",  "Zb",  "Yb",
```
Comment (hadley) [inline R/formatter.r]: Can you please fix the indenting here?

File: R/formatter.r [hunk 2]
```diff
@@ -121,6 +121,89 @@ percent_format <- function() {
...
+				"B",  "KB",  "MB",  "GB",  "TB",  "PB",  "EB",  "ZB",  "YB",
+				     "KiB", "MiB", "GiB", "TiB", "PiB", "EiB", "ZiB", "YiB"))
+
+  units <- match.arg(units, c("binary", "si"))
+
+  base <- switch(units, `binary`=1024, `si`=1000)
```
Comment (hadley) [inline R/formatter.r]: Missing spaces around `=`, and don't need `` ` ``

File: R/formatter.r [hunk 3]
```diff
@@ -121,6 +121,89 @@ percent_format <- function() {
...
+
+  if (symbol == "auto") {
+    symbol <-
```
Comment (hadley) [inline R/formatter.r]: Can you please convert this to `cut(log(x, base))` etc?

File: R/formatter.r [hunk 4]
```diff
@@ -121,6 +121,89 @@ percent_format <- function() {
...
+#'   equivalents and "\code{iB}" equivalents.
+#' @param units which unit base to use, "\code{binary}" (1024 base) or
+#'   "\code{si}" (1000 base) for ISI units.
+#' @references Units of Information (Wikipedia) :
+#'   \url{http://en.wikipedia.org/wiki/Units_of_information}
+#' @export
+#' @examples
+#' byte_format()(sample(3000000000, 10))
+#' bytes(sample(3000000000, 10))
+byte_format <- function(symbol="auto", units="binary") {
```
Comment (hadley) [inline R/formatter.r]: Spaces around `=` please

File: R/formatter.r [hunk 5]
```diff
@@ -121,6 +121,89 @@ percent_format <- function() {
...
+  function(x) bytes(x, symbol, units)
+}
+
+#' @export
+#' @rdname byte_format
+Kb <- byte_format("Kb", "binary")
```
Comment (hadley) [inline R/formatter.r]: I'm always slightly concerned about using very short names for functions that won't be used that commonly.

File: R/formatter.r [hunk 6]
```diff
@@ -121,6 +121,89 @@ percent_format <- function() {
...
+      if      (max(x) >= (base^5)) { "Pb" }
+      else if (max(x) >= (base^4)) { "Tb" }
+      else if (max(x) >= (base^3)) { "Gb" }
+      else if (max(x) >= (base^2)) { "Kb" }
+      else if (max(x) >= (base^1)) { "Mb" }
+      else                         {  "b" }
+  }
+
+  switch(symbol,
+	   "b" =,  "B"  = paste(x,                  "bytes"),
```
Comment (hadley) [inline R/formatter.r]: I'd prefer to do this with character indexing, and to eliminate a lot of the duplicate code. (Maybe just look at the first letter of the unit?)

**Comment by hadley** [comment]
Like the idea, but lots of minor comments. Can you please also merge/rebase to master?

**Comment by hrbrmstr** [comment]
Will do. I can remove the helper functions, too. They're just a convenience
and easy for folks to snap on after if needed.

**Comment by hadley** [comment]
Do you have time to do today? Otherwise it's unlikely to make it for this release.

**Comment by hadley** [comment]
Bump? If you can get it done by the end of next week, I can include it in the next release.

**Comment by hrbrmstr** [comment]
Thx for the bump (this helps me re-prioritize a cpl things). It will be
done by end of weekend, now :-)

**Comment by hadley** [comment]
>2 years later, are you still interested in finishing this off? @dpseidel is here to help 😄

**Comment by hadley** [comment]
One more ping 😄 

@dpseidel can also help out, if needed.

**Comment by dpseidel** [comment]
@hrbrmstr, Can I help finish this off? We've just merged some big changes to the number formatters in scales and this would fit in nicely. Happy to take it over in a fresh PR or help in any way you'd like.

**Pull request closed without merging**